// Bug happens on JVM , JVM -Xuse-ir
// IGNORE_BACKEND_FIR: JVM_IR
// FILE: tmp0.kt

interface A<T> {
    fun foo(OK: T): Any?
}

class Derived(a: A<Int>) : A<Int> by (a!!)

fun box(): (String)? {
    val o = mutableListOf<UInt>()
    
(2.unaryMinus()!! === null)!!.not()
fun tlsa(): Any {
        

Exception()!!
(10!!.unaryMinus() === null)!!.not()!!.not()
if (return "fail get") {
println("""THEN""");
return true!!
} else{
println("""ELSE""");

    
407340732!!.unaryMinus()
(1!!.unaryMinus() != null)!!.not()!!!!.not()
return "step"
}
1.unaryMinus()
}
object {
            val v2 = (10!!.toByte()!! as Byte?)!!.hashCode()
        private val String = listOf("qqxzk")
}


0.unaryMinus().unaryMinus()
fun qbau(): String {
    val x: Long = 5
    if (x != return "OK") {
println("""THEN""");
return "fail 2"!!
}
    
try{
println("""TRY""");

if ((x)) {
println("""THEN""");
if (((false).not()!!.not())) (-6)?.equals(try {

return "OK"


} catch (e: (Throwable)) {
        }
finally{
""

}) == (try {
0
} catch(A: Exception){}
finally{
    val f = "kotlin"::length
    val result = (f.get())
    (return if (result == 5) "OK" else "Fail: $result")
} === null).not()
}
} catch(new: Exception){
println("""CATCH new: Exception""");

}
finally{
println("""FINALLY""");

object : A<Int> {
        override fun foo(t: Int) = if ((t)?.equals(100) ?: ((1).unaryMinus() === null)) {
println("""THEN""");
"OK"
} else {
println("""ELSE""");
"fail 0"
}
    val list4 = ArrayList<ULong>()
}
}!!
    
@file:[JvmName("MultifileClass") JvmMultifileClass]
if (3 != 2) {
println("""THEN""");
return "K"!!
}
0L
    
    
String()!!
2?.equals(try{
println("""TRY""");

fun Int.gxnm() {}

val nbwu = (3u.toUShort()!!..5u.toUShort())!!.reversed()
return "OK"
} catch(a: RuntimeException){
println("""CATCH a: RuntimeException""");

}) ?: (0 === null).not()
(return "OK")
(1385735316.unaryMinus()!! != null)!!.not()
}


return "fail"
}

